networkx/algorithms/traversal/breadth_first_search.py

```diff
@@ -370,6 +371,62 @@ def bfs_successors(G, source, depth_limit=None, sort_neighbors=None):
     yield (parent, children)
 
 
+def bfs_layers(G, sources, layer_limit=None):
+    """Returns all the layes of bfs traversal.
+    Parameters
+    ----------
+    G : NetworkX graph
+        A graph
+
+    source : node in `G` or list of nodes in `G`
+
```

Comment:
Add a sentence describing what this `source` will be used for. Also, it should be named `sources` instead of `source`.

Similarly, for the incoming graph G, you should add a more informative sentence describing G. "A graph" doesn't say how it will be used in the function. Maybe "The graph over which to find the layers using breadth first search."

Thirdly:  Let's make this function a generator.  Instead of returning all the layers at once in a dict, let's have it `yield` the layers one at a time. Then we don't need the depth limit -- the user can stop when they want to. And they can choose whether they want to store it as a list, dict, tuple, or process each layer when it is computed and throw that away before getting the next layer.  Changes needed: remove the `all_layers` dict and replace the setting of that dict with `yield current_layer`. Remove the line at the end `return all_layers`.  Remove the computation of `current_distance` and simplify the loop spec to `while current_layer:`. (Python evaluates a list as "False" when it is empty.) Change the doc_string to replace the "Returns" section with a "Yields" section. The items yielded should be lists of nodes with a description that they are the nodes in each layer of the BFS. In the doc_string examples, you will need to change the calling code to add `list(...)` around the call (or `dict(enumerate(...))` which will give the output you current provide. If you want the slightly simpler output list for these examples, you'll need to change the expected output accordingly.

Style choice:  `visited = {node for node in sources}` can be `visited = set(sources)`... that's up to you. Similar for `current_layer = list(sources)` if you want that.

Finally, the comments in the issue indicate that we should store the layers as lists instead of sets.